File: StudyGamePrograming08/ClearPict.cpp
#include "ClearPict.h"
#include "SpriteComponent.h"
#include "Game.h"
#include "Renderer.h"

ClearPict::ClearPict(Game* game) : Actor(game)
{
	SetState(EPaused);
	SetPosition(-50.0f * Vector3::UnitZ);

	//スプライトコンポーネント作成、テクスチャ設定
	sc = new SpriteComponent(this, 100);
	sc->SetTexture(game->GetRenderer()->GetTexture("Assets/ClearPict.png"));
	sc->SetVisible(false);
}

void ClearPict::UpdateActor(float deltaTime)
{
	int numAsteroids = GetGame()->GetAsteroids().size();
	if (numAsteroids <= 0)
	{
		SetState(EActive);
		sc->SetVisible(true);
	}
}

File: StudyGamePrograming08_VB/ClearPict.vb
Imports OpenTK.Mathematics

Public Class ClearPict
    Inherits Actor

    Private sc As SpriteComponent

    Sub New(ByRef game As Game)
        MyBase.New(game)
        SetState(State.EPaused)
        SetPosition(-50.0 * Vector3.UnitZ)

        'スプライトコンポーネント作成、テクスチャ設定
        sc = New SpriteComponent(Me, 100)
        sc.SetTexture(game.GetRenderer().GetTexture("Assets\ClearPict.png"))
        sc.SetVisible(False)
    End Sub
    Public Overrides Sub UpdateActor(ByVal deltaTime As Double)
        Dim numAsteroids As Integer = GetGame().GetAsteroids().Count()
        If (numAsteroids <= 0) Then
            SetState(State.EActive)
            sc.SetVisible(True)
        End If
    End Sub
End Class